pkg/security/certs.go

```diff
@@ -0,0 +1,336 @@
+	NodePem
+	// UIPem describes the server certificate for the admin UI.
+	UIPem
+	// ClientPem describes a client certificate.
+	ClientPem
+	// TenantClientPem describes a SQL tenant client certificate.
+	TenantClientPem
+
+	// Maximum allowable permissions.
+	maxKeyPermissions os.FileMode = 0700
+	// Filename extenstions.
+	certExtension = `.crt`
+	keyExtension  = `.key`
+	// Certificate directory permissions.
+	defaultCertsDirPerm = 0700
+)
+
+const (
+	CR            string = "cockroach"
+	CERT          string = "cert"
+	CREATE_CA     string = "create-ca"
+	CREATE_NODE   string = "create-node"
+	CREATE_CLIENT string = "create-client"
+
+	CERTS_DIR string = "--certs-dir=%s"
+	CA_KEY    string = "--ca-key=%s"
+)
+
+// CreateCAPair creates a general CA certificate and associated key.
+func CreateCAPair(
+	certsDir, caKeyPath string,
+	keySize int,
+	lifetime time.Duration,
+	allowKeyReuse bool,
+	overwrite bool,
+) error {
+	return createCACertAndKey(certsDir, caKeyPath, CAPem, keySize, lifetime, allowKeyReuse, overwrite)
+}
+
+// CreateClientCAPair creates a client CA certificate and associated key.
+func CreateClientCAPair(
+	certsDir, caKeyPath string,
+	keySize int,
+	lifetime time.Duration,
+	allowKeyReuse bool,
+	overwrite bool,
+) error {
+	return createCACertAndKey(certsDir, caKeyPath, ClientCAPem, keySize, lifetime, allowKeyReuse, overwrite)
+}
+
+// CreateUICAPair creates a UI CA certificate and associated key.
+func CreateUICAPair(certsDir, caKeyPath string, keySize int, lifetime time.Duration, allowKeyReuse bool, overwrite bool) error {
+	return createCACertAndKey(certsDir, caKeyPath, UICAPem, keySize, lifetime, allowKeyReuse, overwrite)
+}
+
+// createCACertAndKey creates a CA key and a CA certificate.
+// If the certs directory does not exist, it is created.
+// If the key does not exist, it is created.
+// The certificate is written to the certs directory. If the file already exists,
+// we append the original certificates to the new certificate.
+//
+// The filename of the certificate file must be specified.
+// It should be one of:
+// - ca.crt: the general CA certificate
+// - ca-client.crt: the CA certificate to verify client certificates
+func createCACertAndKey(certsDir, caKeyPath string, caType PemUsage, keySize int, lifetime time.Duration, allowKeyReuse bool, overwrite bool) error {
+	if len(caKeyPath) == 0 {
+		return errors.New("the path to the CA key is required")
+	}
+	if len(certsDir) == 0 {
+		return errors.New("the path to the certs directory is required")
+	}
+	if caType != CAPem &&
+		caType != TenantClientCAPem &&
+		caType != ClientCAPem &&
+		caType != UICAPem {
+
+		return fmt.Errorf("caType argument to createCACertAndKey must be one of CAPem (%d), ClientCAPem (%d), or UICAPem (%d), got: %d", CAPem, ClientCAPem, UICAPem, caType)
+	}
+
+	certsDirParam := fmt.Sprintf(CERTS_DIR, certsDir)
+	caKeyParam := fmt.Sprintf(CA_KEY, caKeyPath)
+
+	// TODO do we need to move the certs to the correct path?
```

Comment:
does the command we're running in `execCmd` write the certs to a path that the generateCert rc specifics when calling ? 
If so i think we can ignore moving the certs to the correct path? Does the container get destroyed after the generated certs are persisted at k8s secrets?